test/test_binary_ufuncs.py

```diff
@@ -1461,33 +1464,12 @@ def _reference_implementation(x, mod):
                                 low=1, high=9)
         # non-contiguous
         x_nc = x.t()
-        mod_nc = mod.t()
-
-        # Mods: Integer, Float, Tensor, Non-contiguous Tensor
-        mods = [3, 2.3, mod, mod_nc]
-        for m in mods:
-            _reference_implementation(x, m)
-            _reference_implementation(x_nc, m)
-
-        # Integral Tensor fmod to floating-point Tensor
-        # Can not cast floating-point result to original integral Tensor without type promotion
-        if dtype in torch.testing.get_all_int_dtypes():
-            res = torch.fmod(x, mod_float)
-            exp = np.fmod(x.cpu().numpy(), mod_float.cpu().numpy())
-            exp = torch.from_numpy(exp)
-            res = res.to(exp.dtype)
-            self.assertEqual(res, exp)
-            with self.assertRaisesRegex(RuntimeError, "result type (Half|Float|Double) "
-                                                      "can't be cast to the desired "
-                                                      "output type (Byte|Char|Short|Int|Long)"):
-                out = torch.empty(0, device=device, dtype=dtype)
-                torch.fmod(x, mod_float, out=out)
-            with self.assertRaisesRegex(RuntimeError, "result type (Half|Float|Double) "
-                                                      "can't be cast to the desired "
-                                                      "output type (Byte|Char|Short|Int|Long)"):
-                x.fmod_(mod_float)
-        else:
-            _reference_implementation(x, mod_float)
+
+        # Mods: Integer, Float, Tensor, Non-contiguous Tensor, float Tensor
+        mods = [3, 2.3, mod, mod.t(), mod_float]
```

Comment:
I think this enumeration can be improved upon.

If you put all tensors in one tuple, you can put the same tensors + scalars in a second tuple and cross-product them. Then enumerate the elements of the cross-product and run the test.

Also, mod_float should only be created when the dtype is an integer type. One way to do this is just by appending it to the tensor iterable when the dtype is an integer type.